homeassistant/helpers/config_validation.py

```diff
@@ -1769,10 +1771,32 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict[str, Any]:
     )
 )
 
+
+def _backward_compat_trigger_schema(value: Any | None) -> Any:
+    """Backward compatibility for trigger schemas."""
+
+    if not isinstance(value, dict):
+        return value
+
+    response = copy(value)
+
+    # `platform` has been renamed to `trigger`
+    # But we still use `platform` to maintain backward compatibility
```

Comment:
This explanation is not clear, we should explain the `trigger` key is currently only for yaml, while the Python implementation still uses `platform`